// @ng-bootstrap/ng-bootstrap/datepicker/datepicker.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var common_1 = require('@angular/common');
var datepicker_1 = require('./datepicker');
var datepicker_month_view_1 = require('./datepicker-month-view');
var datepicker_navigation_1 = require('./datepicker-navigation');
var forms_1 = require('@angular/forms');
var datepicker_day_view_1 = require('./datepicker-day-view');
var datepicker_i18n_1 = require('./datepicker-i18n');
var ngb_calendar_1 = require('./ngb-calendar');
var datepicker_service_1 = require('./datepicker-service');
var datepicker_navigation_select_1 = require('./datepicker-navigation-select');
var datepicker_config_1 = require('./datepicker-config');
var datepicker_config_2 = require('./datepicker-config');
exports.NgbDatepickerConfig = datepicker_config_2.NgbDatepickerConfig;
var NgbDatepickerModule = (function () {
    function NgbDatepickerModule() {
    }
    NgbDatepickerModule = __decorate([
        core_1.NgModule({
            declarations: [
                datepicker_1.NgbDatepicker, datepicker_month_view_1.NgbDatepickerMonthView, datepicker_navigation_1.NgbDatepickerNavigation, datepicker_navigation_select_1.NgbDatepickerNavigationSelect, datepicker_day_view_1.NgbDatepickerDayView
            ],
            exports: [datepicker_1.NgbDatepicker],
            imports: [common_1.CommonModule, forms_1.FormsModule],
            providers: [
                { provide: ngb_calendar_1.NgbCalendar, useClass: ngb_calendar_1.NgbCalendarGregorian },
                { provide: datepicker_i18n_1.NgbDatepickerI18n, useClass: datepicker_i18n_1.NgbDatepickerI18nDefault }, datepicker_service_1.NgbDatepickerService, datepicker_config_1.NgbDatepickerConfig
            ]
        }), 
        __metadata('design:paramtypes', [])
    ], NgbDatepickerModule);
    return NgbDatepickerModule;
}());
exports.NgbDatepickerModule = NgbDatepickerModule;
